packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -913,6 +932,47 @@ class SlideshowView extends GalleryComponent {
     return {arrowRenderer, navArrowsContainerWidth, navArrowsContainerHeight}
   }
 
+  // Function that checks if the nav arrows parent-container is large enough for them
+  arrowsWillFitPosition = () => {
+    const {
+      isSlideshow,
+      slideshowInfoSize,
+      arrowsVerticalPosition,
+      textBoxHeight,
+    } = this.props.options;
+    const { height } = this.props.container;
+    // Calc of Nav arrows container's height
+    const { navArrowsContainerHeight } = this.getArrowsSizeData();
+    let parentHeight;
+    const infoHeight = isSlideshow ? slideshowInfoSize : textBoxHeight;
+    switch (arrowsVerticalPosition){
+      case GALLERY_CONSTS.arrowsVerticalPosition.INFO_CENTER:
+        parentHeight = infoHeight;
+        break;
+      case GALLERY_CONSTS.arrowsVerticalPosition.IMAGE_CENTER:
+        parentHeight = height - infoHeight;
+        break;
+      case GALLERY_CONSTS.arrowsVerticalPosition.ITEM_CENTER:
+      default:
+        parentHeight = height;
+        break;
+    }
+    return parentHeight >= navArrowsContainerHeight;
+
+  }
+
+  // function to Determine whether we should render the navigation arrows
+  shouldRenderNavArrows = () => {
+    const { showArrows } = this.props.options;
```

Comment:
Please move all the navigation arrows methods you added to a new file. Call it `navigationArrowUtils`. 
The functions should have no dependency in `this`.
Pass each function the relevant arguments it needs.
I want the navigation arrows logic separated from the slideshowView file.  (since HOC refactoring is too much for now, then a simple separation will be enough at the moment)